pkg/metadata/inventories/inventories.go

```diff
@@ -203,6 +210,25 @@ func createPayload(ctx context.Context, hostname string, ac AutoConfigInterface,
 	checkMetadataMutex.Lock()
 	defer checkMetadataMutex.Unlock()
 
+	// setLogLevel select the correct log level for the inventory payload currently being created. We send a new payload
+	// every 1 to 10 min (depending on new metadata being registered). We don't want to log the same error again and again.
+	// We log once every 12 times on normal log level and on debug the rest of the time. The metadata in this paylaod should
+	// not change often, so with 1 paylaod every 10 minutes we would log once every 2h.
+	logLevelMutex.Lock()
+	defer logLevelMutex.Unlock()
```

Comment:
What is this protecting?  It looks like, just `logCount`, since logErrorf and logInfof are read without holding this lock.  Assuming there's concurrency here (why else have mutexes?), that presents a race condition that might allow torn writes.

It might be better to just have an atomic to handle logCount, and access that within (fixed) wrapper functions logErrorf and logInfof.

Reply:
As we discussed I also cleanup the mutexes in the package to have a single one